### ISSUE TITLE
manifolds-0.1.6.3 fails to install

### ISSUE THREAD

**Issue by redneb** [leftaroundabout/manifolds#2]
I just tried to install manifolds-0.1.6.3 but the installation failed with the following message:

```
[17 of 20] Compiling Data.Manifold.TreeCover ( Data/Manifold/TreeCover.hs, dist/dist-sandbox-28c5febe/build/Data/Manifold/TreeCover.o )

Data/Manifold/TreeCover.hs:234:49:
    Couldn't match type ‘[]’
                   with ‘(->) (DualSpace (DualSpace (Interior (Needle (Needle x)))))’
    Expected type: DualSpace (DualSpace (Interior (Needle (Needle x))))
                   -> Metric' x
      Actual type: [Metric' x]
    Relevant bindings include
      cv₂ :: DualSpace (DualSpace (Interior (Needle (Needle x))))
        (bound at Data/Manifold/TreeCover.hs:233:45)
      cv₁ :: DualSpace (DualSpace (Interior (Needle (Needle x))))
        (bound at Data/Manifold/TreeCover.hs:232:45)
      cc :: x (bound at Data/Manifold/TreeCover.hs:231:32)
      b₂ :: Scalar (DualSpace (DualSpace (Interior (Needle (Needle x)))))
        (bound at Data/Manifold/TreeCover.hs:229:14)
      b₁ :: Scalar (DualSpace (DualSpace (Interior (Needle (Needle x)))))
        (bound at Data/Manifold/TreeCover.hs:228:14)
      e₂' :: HerMetric
               (DualSpace (DualSpace (Interior (Needle (Needle x)))))
        (bound at Data/Manifold/TreeCover.hs:227:19)
      (Some bindings suppressed; use -fmax-relevant-binds=N or -fno-max-relevant-binds)
    In the first argument of ‘(<$>)’, namely ‘[e₁, e₂] ++ projector'’
    In the second argument of ‘($)’, namely
      ‘[e₁, e₂] ++ projector' <$> [cv₁, cv₂]’
    In the expression:
      Shade cc . sumV $ [e₁, e₂] ++ projector' <$> [cv₁, cv₂]

Data/Manifold/TreeCover.hs:234:61:
    Couldn't match expected type ‘[Metric' x]’
                with actual type ‘v0 -> HerMetric' v0’
    Relevant bindings include
      cv₂ :: DualSpace (DualSpace (Interior (Needle (Needle x))))
        (bound at Data/Manifold/TreeCover.hs:233:45)
      cv₁ :: DualSpace (DualSpace (Interior (Needle (Needle x))))
        (bound at Data/Manifold/TreeCover.hs:232:45)
      cc :: x (bound at Data/Manifold/TreeCover.hs:231:32)
      b₂ :: Scalar (DualSpace (DualSpace (Interior (Needle (Needle x)))))
        (bound at Data/Manifold/TreeCover.hs:229:14)
      b₁ :: Scalar (DualSpace (DualSpace (Interior (Needle (Needle x)))))
        (bound at Data/Manifold/TreeCover.hs:228:14)
      e₂' :: HerMetric
               (DualSpace (DualSpace (Interior (Needle (Needle x)))))
        (bound at Data/Manifold/TreeCover.hs:227:19)
      (Some bindings suppressed; use -fmax-relevant-binds=N or -fno-max-relevant-binds)
    Probable cause: ‘projector'’ is applied to too few arguments
    In the second argument of ‘(++)’, namely ‘projector'’
    In the first argument of ‘(<$>)’, namely ‘[e₁, e₂] ++ projector'’

Data/Manifold/TreeCover.hs:853:49:
    Couldn't match type ‘[]’ with ‘(->) ([a], [a])’
    Expected type: ([a], [a]) -> ([a], [a])
      Actual type: [([a], [a])]
    Relevant bindings include
      xs :: [a] (bound at Data/Manifold/TreeCover.hs:853:28)
      x :: a (bound at Data/Manifold/TreeCover.hs:853:26)
      partitionsOfFstLength :: Int -> [a] -> [([a], [a])]
        (bound at Data/Manifold/TreeCover.hs:851:1)
    In the second argument of ‘(<$>)’, namely
      ‘partitionsOfFstLength (n - 1) xs ++ second (x :)’
    In the first argument of ‘(<$>)’, namely
      ‘first (x :) <$> partitionsOfFstLength (n - 1) xs ++ second (x :)’
    In the expression:
      first (x :) <$> partitionsOfFstLength (n - 1) xs ++ second (x :)
      <$> partitionsOfFstLength n xs

Data/Manifold/TreeCover.hs:854:34:
    Couldn't match type ‘(->) ([a], [a])’ with ‘[]’
    Expected type: [([a], [a])]
      Actual type: ([a], [a]) -> ([a], [a])
    Relevant bindings include
      xs :: [a] (bound at Data/Manifold/TreeCover.hs:853:28)
      x :: a (bound at Data/Manifold/TreeCover.hs:853:26)
      partitionsOfFstLength :: Int -> [a] -> [([a], [a])]
        (bound at Data/Manifold/TreeCover.hs:851:1)
    Probable cause: ‘second’ is applied to too few arguments
    In the second argument of ‘(++)’, namely ‘second (x :)’
    In the second argument of ‘(<$>)’, namely
      ‘partitionsOfFstLength (n - 1) xs ++ second (x :)’
Failed to install manifolds-0.1.6.3
```

This happened on a brand new cabal sandbox. I am on ghc 7.10.3

**Comment by leftaroundabout** [comment]
Thanks for the feedback. I clearly messed something up there. Not sure whether it's worth digging into the cause though &ndash; I just uploaded [a new version](http://hackage.haskell.org/package/manifolds-0.2.0.1) to hackage, that should work.

**Comment by leftaroundabout** [comment]
Ok, I found out what went wrong. I made some changes in `constrained-categories-0.2.5`; it turns out these were _breaking changes_ for the old `manifolds` versions (so I probably [should have bumped the major version number](https://wiki.haskell.org/Package_versioning_policy)).

As a workaround, I now restrict the dependency in the `manifolds-0.1.*` versions to `constrained-categories<0.2.5` on hackage, so after `cabal update` all these versions should install again now.
